chew.go

```diff
@@ -1,18 +1,54 @@
+/*
+Package chew provides a simple way to process URLs and files. It allows you to process a list of URLs
+and files and returns the content of the URLs and files as a list of Chunks. It also provides a way to
+transcribe audio files using the Google Cloud Speech-to-Text API or the OpenAI Whisper API.
+
+The library respects robots.txt files and crawl delays, and allows you to set a custom http.Client for making requests.
+
+Note on Responsible Usage:
+
+This library is designed for processing data from both local files and web sources. Users should be aware of the following considerations:
+
+1. Web Scraping:
+  - When scraping websites, ensure compliance with the target website's terms of service and robots.txt rules.
+  - Respect rate limits and crawl delays to avoid overwhelming target servers.
+  - Be aware that web scraping may be subject to legal restrictions in some jurisdictions.
+
+2. File Processing:
+  - Exercise caution when processing files from untrusted sources.
+  - Ensure you have appropriate permissions to access and process the files.
+  - Be mindful of potential sensitive information in processed files and handle it securely.
+
+3. Data Handling:
+  - Properly secure and manage any data extracted or processed using this library, especially if it contains personal or sensitive information.
+  - Comply with relevant data protection regulations (e.g., GDPR, CCPA) when handling personal data.
+
+4. System Resource Usage:
+  - Be aware that processing large files or numerous web pages can be resource-intensive. Monitor and manage system resources accordingly.
+
+5. Have Fun
+
+Users of this library are responsible for ensuring their usage complies with applicable laws, regulations, and ethical considerations in their jurisdiction and context of use.
+*/
 package chew
 
 import (
 	"context"
 	"fmt"
 	"io"
 	"net/http"
+	"net/url"
 	"strings"
 	"sync"
+	"time"
 
 	"github.com/mmatongo/chew/v1/internal/common"
 	"github.com/mmatongo/chew/v1/internal/document"
 	"github.com/mmatongo/chew/v1/internal/text"
 	"github.com/mmatongo/chew/v1/internal/transcribe"
 	"github.com/mmatongo/chew/v1/internal/utils"
+	"github.com/temoto/robotstxt"
+	"golang.org/x/time/rate"
```

Comment:
Usually external libraries are added in a dedicated section


```suggestion
 	"context"
 	"fmt"
 	"io"
 	"net/http"
	"net/url"
 	"strings"
 	"sync"
	"time"

	"github.com/temoto/robotstxt"
	"golang.org/x/time/rate"
 
 	"github.com/mmatongo/chew/v1/internal/common"
 	"github.com/mmatongo/chew/v1/internal/document"
 	"github.com/mmatongo/chew/v1/internal/text"
 	"github.com/mmatongo/chew/v1/internal/transcribe"
 	"github.com/mmatongo/chew/v1/internal/utils"
```

Reply:
You didn't work on this